storey/table.py

```diff
@@ -1162,13 +1162,24 @@ def get_nearest_window_index_by_timestamp(self, timestamp, window_millis):
     def remove_old_values_from_pre_aggregations(self, timestamp):
         if self._precalculated_aggregations:
             for (aggr_name, current_window_millis), aggr in self._current_aggregate_values.items():
-                previous_window_start = self.get_window_range(self.get_end_bucket(self._last_data_point_timestamp), current_window_millis)
-                current_window_start = self.get_window_range(self.get_end_bucket(timestamp), current_window_millis)
+                self.is_fixed_window = isinstance(self.explicit_windows, FixedWindows)
+                if self.is_fixed_window:
+                    previous_window_start_time = \
+                        self.get_window_start_time_from_timestamp(self._last_data_point_timestamp, current_window_millis)
+                    previous_window_start = self.get_bucket_index_by_timestamp(previous_window_start_time)
+                    current_window_start_time = \
+                        self.get_window_start_time_from_timestamp(timestamp, current_window_millis)
+                    current_window_start = self.get_bucket_index_by_timestamp(current_window_start_time)
+                else:
+                    previous_window_start = \
+                        self.get_window_range(self.get_end_bucket(self._last_data_point_timestamp), current_window_millis)
+                    current_window_start = \
+                        self.get_window_range(self.get_end_bucket(timestamp), current_window_millis)
 
-                previous_window_start = max(0, previous_window_start)
-                current_window_start = max(0, current_window_start)
-                previous_window_start = min(len(self.buckets) - 1, previous_window_start)
-                current_window_start = min(len(self.buckets), current_window_start)
+                    previous_window_start = max(0, previous_window_start)
+                    current_window_start = max(0, current_window_start)
+                    previous_window_start = min(len(self.buckets) - 1, previous_window_start)
+                    current_window_start = min(len(self.buckets), current_window_start)
```

Comment:
The min here does seem right... Can this actually happen? Would it be correct?

Reply:
same as above